### PR TITLE
Feat: convert bls sdk to abstract contract which has base included

### PR DESCRIPTION
**Pull request by rishabh-vasudevan19**

### CODE REVIEW

File: src/entities/BlsBaseDSSLib.sol [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+library BlsBaseDSSLib {
+    using BN254 for BN254.G1Point;
+
+    ///@notice the user of the library should initiate a storage variable that has been built on this struct
+    struct State {
+        ICore core;
+        mapping(address operatorAddress => bool exists) operatorExists;
+        mapping(address operatorAddress => uint256 index) operatorIndex;
+        address[] operatorAddresses;
```
Comment (devdks25) [inline src/entities/BlsBaseDSSLib.sol:16]: Use `BaseDSSLib.State` here instead and overload `baseDssStatePtr()` pointing to this.

File: src/BlsBaseDSS.sol [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+import "forge-std/console.sol";
+import {IBaseDSS} from "./interfaces/IBaseDSS.sol";
+import {Constants} from "./interfaces/Constants.sol";
+import {BN254} from "./entities/BN254.sol";
+import {ICore} from "./interfaces/ICore.sol";
+import {BaseDSSOperatorLib} from "./entities/BaseDSSOperatorLib.sol";
+import {BlsBaseDSSLib} from "./entities/BlsBaseDSSLib.sol";
+import {IStakeViewer} from "./interfaces/IStakeViewer.sol";
+
+abstract contract BlsBaseDSS is IBaseDSS {
```
Comment (devdks25) [inline src/BlsBaseDSS.sol:13]: Import `BaseDSS` here.

Comment (rishabh-vasudevan19) [reply src/BlsBaseDSS.sol:13]: Need a different state than BaseDSS, that is why have created a parallel to baseDSS for DSS's that want to build on BLS.

File: src/entities/BlsBaseDSSLib.sol (hunk 1 above)
Comment (rishabh-vasudevan19) [reply src/entities/BlsBaseDSSLib.sol:15]: Need different state than BaseDSS, that's why made a new one

File: src/BlsBaseDSS.sol [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    }
+
+    /* ============ Internal Functions ============ */
+
+    /**
+     * @notice initializes the DSS
+     * @param core the core contract address
+     * @param maxSlashablePercentageWad the maximum percentage of the stake that can be slashed
+     */
+    function init(
```
[the remark this answers is not in the document]
Comment (rishabh-vasudevan19) [reply src/BlsBaseDSS.sol:185]: yeah will take a look and update to that

**Pull request merged**